Crash when seasons marked as played

### Describe the bug

1. Go to a show
2. Go to "Seasons"
3. Mark a season as played
4. ASAP; mark another season as played

**Expected:**
No crash and both seasons are marked as played in the UI

**Actual:**
At step 3 the season is not marked as played in the UI
At step 4 it crashed almost everytime (at least I'm moved to the Android TV launcher app) - guess it depends on how fast you can perform step 3 and 4 and maybe how fast hardware you got. After going back into Jellyfin both seasons are correctly marked as played.


### Logs

_No response_

### Application version

0.16.2

### Where did you install the app from?

Google Play

### Device information

BenQ QS02

### Android version

11

### Jellyfin server version

10.8.13

I did manage to crash it this time and get a `upload_org.jellyfin.androidtv...` file but this time I did something a bit dissimilar. I marked 5 seasons as "Unplayed" as fast as I could and then backed out of the show screen and then back in. After opening Jellyfin again and going back into the show only 2/5 seasons were marked "Unplayed". Maybe it should be reported as a separate bug? I guess it's related to slow server response as you hinted at before.

---
client: Jellyfin for Android TV
client_version: 0.0.0-dev.1
client_repository: https://github.com/jellyfin/jellyfin-androidtv
type: crash_report
format: markdown
---
### Logs

***Stack Trace***: 
```log
org.jellyfin.sdk.api.client.exception.TimeoutException: HTTP request timed out
	at org.jellyfin.sdk.api.client.KtorClient.request$suspendImpl(KtorClient.kt:117)
	at org.jellyfin.sdk.api.client.KtorClient$request$1.invokeSuspend(Unknown Source:19)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@9459ad5, Dispatchers.Main.immediate]
Caused by: io.ktor.client.features.HttpRequestTimeoutException: Request timeout has expired [url=http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/82ba56a2-f735-c95a-7193-9f340b4d0bc8, request_timeout=30000 ms]
	at io.ktor.client.features.HttpTimeout$Feature$install$1$1$killer$1.invokeSuspend(HttpTimeout.kt:149)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)

```  
***Logcat***: 
```log
02-16 22:43:44.901 D/KeyProcessor(18522): Menu for: Season 4
02-16 22:43:45.028 W/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed
02-16 22:43:46.565 I/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/8664630c-65e9-54c0-87c4-b822b3ea2d7b
02-16 22:43:46.749 W/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed
02-16 22:43:47.737 D/KeyProcessor(18522): Menu for: Season 3
02-16 22:43:47.861 W/libEGL  (18522): EGLNativeWindowType 0xbeccfbf8 disconnect failed
02-16 22:43:49.888 I/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/51e44eb3-5c25-2ec8-b1cc-e9f30e28b861
02-16 22:43:50.086 W/libEGL  (18522): EGLNativeWindowType 0xbeccfbf8 disconnect failed
02-16 22:43:51.087 D/KeyProcessor(18522): Menu for: Season 2
02-16 22:43:51.209 W/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed
02-16 22:43:52.692 I/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/0dc777f4-a239-9817-2889-f69fe0096735
02-16 22:43:52.885 W/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed
02-16 22:43:53.915 D/KeyProcessor(18522): Menu for: Season 1
02-16 22:43:54.054 W/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed
02-16 22:43:55.554 I/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/3be1eb2d-65e4-1be0-35e6-91ed38e3c165
02-16 22:43:55.739 W/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed
02-16 22:43:56.232 D/NavigationRepositoryImpl(18522): Navigating back
02-16 22:43:56.289 D/jellyfin-apiclient(18522): Adding request to queue: http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/Items?ParentId=a656b907-eb3a-7353-2e40-e44b968d0225&SortBy=SortName&recursive=true&Fields=PrimaryImageAspectRatio%2CChildCount%2CMediaSources%2CMediaStreams%2CDisplayPreferencesId&sortOrder=Ascending&Limit=55&IncludeItemTypes=Series&format=json
02-16 22:43:56.369 D/BrowseGridFragment(18522): Refresh item "Arrested Development"
02-16 22:43:56.373 I/org.jellyfin.sdk.api.client.KtorClient(18522): GET http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/Items/49e96fa8-3290-1327-05d4-3f487dba20e9
02-16 22:43:56.398 D/BrowseGridFragment(18522): row selected position 3
02-16 22:43:57.132 I/Choreographer(18522): Skipped 44 frames!  The application may be doing too much work on its main thread.
02-16 22:43:57.149 I/OpenGLRenderer(18522): Davey! duration=862ms; Flags=0, IntendedVsync=12483804161020, Vsync=12483887577685, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=12483891526784, AnimationStart=12483891529367, PerformTraversalsStart=12483891728284, DrawStart=12484616674868, SyncQueued=12484642983576, SyncStart=12484643418368, IssueDrawCommandsStart=12484646132909, SwapBuffers=12484663765076, FrameCompleted=12484667072451, DequeueBufferDuration=319333, QueueBufferDuration=541083, GpuCompleted=12467205236735, 
02-16 22:43:57.179 I/OpenGLRenderer(18522): Davey! duration=792ms; Flags=0, IntendedVsync=12483904260746, Vsync=12484638327398, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=12484649899618, AnimationStart=12484649904701, PerformTraversalsStart=12484650270743, DrawStart=12484663094909, SyncQueued=12484669616451, SyncStart=12484669964743, IssueDrawCommandsStart=12484672167618, SwapBuffers=12484693572493, FrameCompleted=12484696721368, DequeueBufferDuration=268042, QueueBufferDuration=2336500, GpuCompleted=12467235355901, 
02-16 22:43:57.235 I/androidtv.debu(18522): Background young concurrent copying GC freed 164691(6693KB) AllocSpace objects, 18(360KB) LOS objects, 28% free, 18MB/25MB, paused 224us total 149.120ms
02-16 22:43:57.239 D/BrowseGridFragment(18522): Refresh item "Arrested Development"
02-16 22:43:57.241 I/org.jellyfin.sdk.api.client.KtorClient(18522): GET http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/Items/49e96fa8-3290-1327-05d4-3f487dba20e9
02-16 22:43:58.457 D/BrowseGridFragment(18522): row selected position 4
02-16 22:43:59.463 D/BrowseGridFragment(18522): row selected position 3
02-16 22:44:00.335 D/ItemLauncher(18522): Item selected: 3 - Arrested Development (Series)
02-16 22:44:00.336 D/NavigationRepositoryImpl(18522): Navigating to Fragment(fragment=class org.jellyfin.androidtv.ui.itemdetail.FullDetailsFragment (Kotlin reflection is not available), arguments=Bundle[{ItemId=49e96fa8-3290-1327-05d4-3f487dba20e9}]) (via navigate function)
02-16 22:44:00.375 D/jellyfin-apiclient(18522): Adding request to queue: http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/Items/49e96fa8-3290-1327-05d4-3f487dba20e9?format=json
02-16 22:44:00.410 W/RecyclerView(18522): No adapter attached; skipping layout
02-16 22:44:01.159 D/FullDetailsFragment(18522): current time 1708119841155 last playback event time 0 last refresh time 1708119840375
02-16 22:44:06.351 D/Volley  (18522): [644] NetworkUtility.logSlowRequests: HTTP response for request=<[ ] http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/Items/49e96fa8-3290-1327-05d4-3f487dba20e9?format=json 0x6768b619 NORMAL 32> [lifetime=5975], [size=6187], [rc=200], [retryCount=0]
02-16 22:44:06.354 I/jellyfin-apiclient(18522): Response received from: http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/Items/49e96fa8-3290-1327-05d4-3f487dba20e9?format=json
02-16 22:44:06.458 D/FullDetailsFragment(18522): Item type: Series
02-16 22:44:06.460 D/jellyfin-apiclient(18522): Adding request to queue: http://<obfuscated_ip>:<obfuscated_port>/Shows/NextUp?Fields=PrimaryImageAspectRatio%2CChildCount&SeriesId=49e96fa83290132705d43f487dba20e9&UserId=<obfuscated_user_id>&format=json
02-16 22:44:06.461 D/jellyfin-apiclient(18522): Adding request to queue: http://<obfuscated_ip>:<obfuscated_port>/Shows/49e96fa83290132705d43f487dba20e9/Seasons?Fields=PrimaryImageAspectRatio%2CDisplayPreferencesId%2CChildCount&UserId=<obfuscated_user_id>&format=json
02-16 22:44:06.463 D/jellyfin-apiclient(18522): Adding request to queue: http://<obfuscated_ip>:<obfuscated_port>/Shows/Upcoming?Fields=PrimaryImageAspectRatio%2CChildCount&UserId=<obfuscated_user_id>&format=json
02-16 22:44:06.467 D/jellyfin-apiclient(18522): Adding request to queue: http://<obfuscated_ip>:<obfuscated_port>/Items/49e96fa83290132705d43f487dba20e9/Similar?Fields=PrimaryImageAspectRatio%2CDisplayPreferencesId%2CChildCount&UserId=<obfuscated_user_id>&Limit=20&format=json
02-16 22:44:09.153 I/jellyfin-apiclient(18522): Response received from: http://<obfuscated_ip>:<obfuscated_port>/Shows/Upcoming?Fields=PrimaryImageAspectRatio%2CChildCount&UserId=<obfuscated_user_id>&format=json
02-16 22:44:12.005 D/Volley  (18522): [645] NetworkUtility.logSlowRequests: HTTP response for request=<[ ] http://<obfuscated_ip>:<obfuscated_port>/Shows/NextUp?Fields=PrimaryImageAspectRatio%2CChildCount&SeriesId=49e96fa83290132705d43f487dba20e9&UserId=<obfuscated_user_id>&format=json 0x6768b619 NORMAL 33> [lifetime=5543], [size=48], [rc=200], [retryCount=0]
02-16 22:44:12.011 I/jellyfin-apiclient(18522): Response received from: http://<obfuscated_ip>:<obfuscated_port>/Shows/NextUp?Fields=PrimaryImageAspectRatio%2CChildCount&SeriesId=49e96fa83290132705d43f487dba20e9&UserId=<obfuscated_user_id>&format=json
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): HTTP request timed out
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): io.ktor.client.features.HttpRequestTimeoutException: Request timeout has expired [url=http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/82ba56a2-f735-c95a-7193-9f340b4d0bc8, request_timeout=30000 ms]
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at io.ktor.client.features.HttpTimeout$Feature$install$1$1$killer$1.invokeSuspend(HttpTimeout.kt:149)
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
02-16 22:44:13.793 D/org.jellyfin.sdk.api.client.KtorClient(18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
02-16 22:44:13.811 E/ACRA    (18522): ACRA caught a TimeoutException for org.jellyfin.androidtv.debug
02-16 22:44:13.811 E/ACRA    (18522): org.jellyfin.sdk.api.client.exception.TimeoutException: HTTP request timed out
02-16 22:44:13.811 E/ACRA    (18522): 	at org.jellyfin.sdk.api.client.KtorClient.request$suspendImpl(KtorClient.kt:117)
02-16 22:44:13.811 E/ACRA    (18522): 	at org.jellyfin.sdk.api.client.KtorClient$request$1.invokeSuspend(Unknown Source:19)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
02-16 22:44:13.811 E/ACRA    (18522): 	at android.os.Handler.handleCallback(Handler.java:938)
02-16 22:44:13.811 E/ACRA    (18522): 	at android.os.Handler.dispatchMessage(Handler.java:99)
02-16 22:44:13.811 E/ACRA    (18522): 	at android.os.Looper.loop(Looper.java:223)
02-16 22:44:13.811 E/ACRA    (18522): 	at android.app.ActivityThread.main(ActivityThread.java:7656)
02-16 22:44:13.811 E/ACRA    (18522): 	at java.lang.reflect.Method.invoke(Native Method)
02-16 22:44:13.811 E/ACRA    (18522): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
02-16 22:44:13.811 E/ACRA    (18522): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
02-16 22:44:13.811 E/ACRA    (18522): 	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@9459ad5, Dispatchers.Main.immediate]
02-16 22:44:13.811 E/ACRA    (18522): Caused by: io.ktor.client.features.HttpRequestTimeoutException: Request timeout has expired [url=http://<obfuscated_ip>:<obfuscated_port>/Users/<obfuscated_user_id>/PlayedItems/82ba56a2-f735-c95a-7193-9f340b4d0bc8, request_timeout=30000 ms]
02-16 22:44:13.811 E/ACRA    (18522): 	at io.ktor.client.features.HttpTimeout$Feature$install$1$1$killer$1.invokeSuspend(HttpTimeout.kt:149)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
02-16 22:44:13.811 E/ACRA    (18522): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
02-16 22:44:13.811 D/ACRA    (18522): Building report
02-16 22:44:13.815 D/ACRA    (18522): Starting collectors with priority FIRST
02-16 22:44:13.816 D/ACRA    (18522): Calling collector org.acra.collector.DropBoxCollector
02-16 22:44:13.816 D/ACRA    (18522): Calling collector org.acra.collector.LogCatCollector
02-16 22:44:13.816 D/ACRA    (18522): Collector org.acra.collector.DropBoxCollector completed
02-16 22:44:13.817 D/ACRA    (18522): Calling collector org.acra.collector.StacktraceCollector
02-16 22:44:13.819 D/ACRA    (18522): Collector org.acra.collector.StacktraceCollector completed
02-16 22:44:13.826 D/ACRA    (18522): Retrieving logcat output (buffer:default)...

```  

### App information

***App version***: `0.0.0-dev.1` (`1`)  
***Package name***: `org.jellyfin.androidtv.debug`  
***Build***: 
```json
{"BOARD":"oppen","BOOTLOADER":"01.01.221026.231954","BRAND":"BenQ","CPU_ABI":"armeabi-v7a","CPU_ABI2":"armeabi","DEVICE":"YUL","DISPLAY":"20221114:user release-keys","FINGERPRINT":"BenQ\/QS02\/YUL:11\/RT\/20221114:user\/release-keys","HARDWARE":"amlogic","HOST":"ubuntu","ID":"RT","IS_DEBUGGABLE":false,"IS_EMULATOR":false,"MANUFACTURER":"SDMC","MODEL":"QS02","PERMISSIONS_REVIEW_REQUIRED":true,"PRODUCT":"QS02","RADIO":"unknown","SUPPORTED_32_BIT_ABIS":["armeabi-v7a","armeabi"],"SUPPORTED_64_BIT_ABIS":[],"SUPPORTED_ABIS":["armeabi-v7a","armeabi"],"TAGS":"release-keys","TIME":1668516221000,"TYPE":"user","UNKNOWN":"unknown","USER":"jiangruifeng","VERSION":{"ACTIVE_CODENAMES":[],"BASE_OS":"","CODENAME":"REL","INCREMENTAL":"20221114","PREVIEW_SDK_FINGERPRINT":"REL","PREVIEW_SDK_INT":0,"RELEASE":"11","RELEASE_OR_CODENAME":"11","SDK":"30","SDK_INT":30,"SECURITY_PATCH":"2022-10-05"}}
```  
***Build config***: 
```json
{"APPLICATION_ID":"org.jellyfin.androidtv.debug","BUILD_TYPE":"debug","DEBUG":true,"DEVELOPMENT":true,"VERSION_CODE":1,"VERSION_NAME":"0.0.0-dev.1"}
```  

### Device information

***Android version***: `11`  
***Device brand***: `BenQ`  
***Device product***: `QS02`  
***Device model***: `QS02`  

### Crash information

***Start time***: `2024-02-16T22:35:57.078+01:00`  
***Crash time***: `2024-02-16T22:44:13.875+01:00`  

### Dump


```json
{"STACK_TRACE":"org.jellyfin.sdk.api.client.exception.TimeoutException: HTTP request timed out\n\tat org.jellyfin.sdk.api.client.KtorClient.request$suspendImpl(KtorClient.kt:117)\n\tat org.jellyfin.sdk.api.client.KtorClient$request$1.invokeSuspend(Unknown Source:19)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n\tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n\tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n\tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n\tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n\tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n\tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)\n\tat android.os.Handler.handleCallback(Handler.java:938)\n\tat android.os.Handler.dispatchMessage(Handler.java:99)\n\tat android.os.Looper.loop(Looper.java:223)\n\tat android.app.ActivityThread.main(ActivityThread.java:7656)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)\n\tSuppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@9459ad5, Dispatchers.Main.immediate]\nCaused by: io.ktor.client.features.HttpRequestTimeoutException: Request timeout has expired [url=http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/82ba56a2-f735-c95a-7193-9f340b4d0bc8, request_timeout=30000 ms]\n\tat io.ktor.client.features.HttpTimeout$Feature$install$1$1$killer$1.invokeSuspend(HttpTimeout.kt:149)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)\n\tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)\n","LOGCAT":"02-16 22:43:44.901 D\/KeyProcessor(18522): Menu for: Season 4\n02-16 22:43:45.028 W\/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed\n02-16 22:43:46.565 I\/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/8664630c-65e9-54c0-87c4-b822b3ea2d7b\n02-16 22:43:46.749 W\/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed\n02-16 22:43:47.737 D\/KeyProcessor(18522): Menu for: Season 3\n02-16 22:43:47.861 W\/libEGL  (18522): EGLNativeWindowType 0xbeccfbf8 disconnect failed\n02-16 22:43:49.888 I\/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/51e44eb3-5c25-2ec8-b1cc-e9f30e28b861\n02-16 22:43:50.086 W\/libEGL  (18522): EGLNativeWindowType 0xbeccfbf8 disconnect failed\n02-16 22:43:51.087 D\/KeyProcessor(18522): Menu for: Season 2\n02-16 22:43:51.209 W\/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed\n02-16 22:43:52.692 I\/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/0dc777f4-a239-9817-2889-f69fe0096735\n02-16 22:43:52.885 W\/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed\n02-16 22:43:53.915 D\/KeyProcessor(18522): Menu for: Season 1\n02-16 22:43:54.054 W\/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed\n02-16 22:43:55.554 I\/org.jellyfin.sdk.api.client.KtorClient(18522): DELETE http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/3be1eb2d-65e4-1be0-35e6-91ed38e3c165\n02-16 22:43:55.739 W\/libEGL  (18522): EGLNativeWindowType 0xbecd79e8 disconnect failed\n02-16 22:43:56.232 D\/NavigationRepositoryImpl(18522): Navigating back\n02-16 22:43:56.289 D\/jellyfin-apiclient(18522): Adding request to queue: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/Items?ParentId=a656b907-eb3a-7353-2e40-e44b968d0225&SortBy=SortName&recursive=true&Fields=PrimaryImageAspectRatio%2CChildCount%2CMediaSources%2CMediaStreams%2CDisplayPreferencesId&sortOrder=Ascending&Limit=55&IncludeItemTypes=Series&format=json\n02-16 22:43:56.369 D\/BrowseGridFragment(18522): Refresh item \"Arrested Development\"\n02-16 22:43:56.373 I\/org.jellyfin.sdk.api.client.KtorClient(18522): GET http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/Items\/49e96fa8-3290-1327-05d4-3f487dba20e9\n02-16 22:43:56.398 D\/BrowseGridFragment(18522): row selected position 3\n02-16 22:43:57.132 I\/Choreographer(18522): Skipped 44 frames!  The application may be doing too much work on its main thread.\n02-16 22:43:57.149 I\/OpenGLRenderer(18522): Davey! duration=862ms; Flags=0, IntendedVsync=12483804161020, Vsync=12483887577685, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=12483891526784, AnimationStart=12483891529367, PerformTraversalsStart=12483891728284, DrawStart=12484616674868, SyncQueued=12484642983576, SyncStart=12484643418368, IssueDrawCommandsStart=12484646132909, SwapBuffers=12484663765076, FrameCompleted=12484667072451, DequeueBufferDuration=319333, QueueBufferDuration=541083, GpuCompleted=12467205236735, \n02-16 22:43:57.179 I\/OpenGLRenderer(18522): Davey! duration=792ms; Flags=0, IntendedVsync=12483904260746, Vsync=12484638327398, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=12484649899618, AnimationStart=12484649904701, PerformTraversalsStart=12484650270743, DrawStart=12484663094909, SyncQueued=12484669616451, SyncStart=12484669964743, IssueDrawCommandsStart=12484672167618, SwapBuffers=12484693572493, FrameCompleted=12484696721368, DequeueBufferDuration=268042, QueueBufferDuration=2336500, GpuCompleted=12467235355901, \n02-16 22:43:57.235 I\/androidtv.debu(18522): Background young concurrent copying GC freed 164691(6693KB) AllocSpace objects, 18(360KB) LOS objects, 28% free, 18MB\/25MB, paused 224us total 149.120ms\n02-16 22:43:57.239 D\/BrowseGridFragment(18522): Refresh item \"Arrested Development\"\n02-16 22:43:57.241 I\/org.jellyfin.sdk.api.client.KtorClient(18522): GET http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/Items\/49e96fa8-3290-1327-05d4-3f487dba20e9\n02-16 22:43:58.457 D\/BrowseGridFragment(18522): row selected position 4\n02-16 22:43:59.463 D\/BrowseGridFragment(18522): row selected position 3\n02-16 22:44:00.335 D\/ItemLauncher(18522): Item selected: 3 - Arrested Development (Series)\n02-16 22:44:00.336 D\/NavigationRepositoryImpl(18522): Navigating to Fragment(fragment=class org.jellyfin.androidtv.ui.itemdetail.FullDetailsFragment (Kotlin reflection is not available), arguments=Bundle[{ItemId=49e96fa8-3290-1327-05d4-3f487dba20e9}]) (via navigate function)\n02-16 22:44:00.375 D\/jellyfin-apiclient(18522): Adding request to queue: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/Items\/49e96fa8-3290-1327-05d4-3f487dba20e9?format=json\n02-16 22:44:00.410 W\/RecyclerView(18522): No adapter attached; skipping layout\n02-16 22:44:01.159 D\/FullDetailsFragment(18522): current time 1708119841155 last playback event time 0 last refresh time 1708119840375\n02-16 22:44:06.351 D\/Volley  (18522): [644] NetworkUtility.logSlowRequests: HTTP response for request=<[ ] http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/Items\/49e96fa8-3290-1327-05d4-3f487dba20e9?format=json 0x6768b619 NORMAL 32> [lifetime=5975], [size=6187], [rc=200], [retryCount=0]\n02-16 22:44:06.354 I\/jellyfin-apiclient(18522): Response received from: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/Items\/49e96fa8-3290-1327-05d4-3f487dba20e9?format=json\n02-16 22:44:06.458 D\/FullDetailsFragment(18522): Item type: Series\n02-16 22:44:06.460 D\/jellyfin-apiclient(18522): Adding request to queue: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Shows\/NextUp?Fields=PrimaryImageAspectRatio%2CChildCount&SeriesId=49e96fa83290132705d43f487dba20e9&UserId=<obfuscated_user_id>&format=json\n02-16 22:44:06.461 D\/jellyfin-apiclient(18522): Adding request to queue: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Shows\/49e96fa83290132705d43f487dba20e9\/Seasons?Fields=PrimaryImageAspectRatio%2CDisplayPreferencesId%2CChildCount&UserId=<obfuscated_user_id>&format=json\n02-16 22:44:06.463 D\/jellyfin-apiclient(18522): Adding request to queue: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Shows\/Upcoming?Fields=PrimaryImageAspectRatio%2CChildCount&UserId=<obfuscated_user_id>&format=json\n02-16 22:44:06.467 D\/jellyfin-apiclient(18522): Adding request to queue: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Items\/49e96fa83290132705d43f487dba20e9\/Similar?Fields=PrimaryImageAspectRatio%2CDisplayPreferencesId%2CChildCount&UserId=<obfuscated_user_id>&Limit=20&format=json\n02-16 22:44:09.153 I\/jellyfin-apiclient(18522): Response received from: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Shows\/Upcoming?Fields=PrimaryImageAspectRatio%2CChildCount&UserId=<obfuscated_user_id>&format=json\n02-16 22:44:12.005 D\/Volley  (18522): [645] NetworkUtility.logSlowRequests: HTTP response for request=<[ ] http:\/\/<obfuscated_ip>:<obfuscated_port>\/Shows\/NextUp?Fields=PrimaryImageAspectRatio%2CChildCount&SeriesId=49e96fa83290132705d43f487dba20e9&UserId=<obfuscated_user_id>&format=json 0x6768b619 NORMAL 33> [lifetime=5543], [size=48], [rc=200], [retryCount=0]\n02-16 22:44:12.011 I\/jellyfin-apiclient(18522): Response received from: http:\/\/<obfuscated_ip>:<obfuscated_port>\/Shows\/NextUp?Fields=PrimaryImageAspectRatio%2CChildCount&SeriesId=49e96fa83290132705d43f487dba20e9&UserId=<obfuscated_user_id>&format=json\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): HTTP request timed out\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): io.ktor.client.features.HttpRequestTimeoutException: Request timeout has expired [url=http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/82ba56a2-f735-c95a-7193-9f340b4d0bc8, request_timeout=30000 ms]\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat io.ktor.client.features.HttpTimeout$Feature$install$1$1$killer$1.invokeSuspend(HttpTimeout.kt:149)\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)\n02-16 22:44:13.793 D\/org.jellyfin.sdk.api.client.KtorClient(18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)\n02-16 22:44:13.811 E\/ACRA    (18522): ACRA caught a TimeoutException for org.jellyfin.androidtv.debug\n02-16 22:44:13.811 E\/ACRA    (18522): org.jellyfin.sdk.api.client.exception.TimeoutException: HTTP request timed out\n02-16 22:44:13.811 E\/ACRA    (18522): \tat org.jellyfin.sdk.api.client.KtorClient.request$suspendImpl(KtorClient.kt:117)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat org.jellyfin.sdk.api.client.KtorClient$request$1.invokeSuspend(Unknown Source:19)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat android.os.Handler.handleCallback(Handler.java:938)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat android.os.Handler.dispatchMessage(Handler.java:99)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat android.os.Looper.loop(Looper.java:223)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat android.app.ActivityThread.main(ActivityThread.java:7656)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat java.lang.reflect.Method.invoke(Native Method)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)\n02-16 22:44:13.811 E\/ACRA    (18522): \tSuppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@9459ad5, Dispatchers.Main.immediate]\n02-16 22:44:13.811 E\/ACRA    (18522): Caused by: io.ktor.client.features.HttpRequestTimeoutException: Request timeout has expired [url=http:\/\/<obfuscated_ip>:<obfuscated_port>\/Users\/<obfuscated_user_id>\/PlayedItems\/82ba56a2-f735-c95a-7193-9f340b4d0bc8, request_timeout=30000 ms]\n02-16 22:44:13.811 E\/ACRA    (18522): \tat io.ktor.client.features.HttpTimeout$Feature$install$1$1$killer$1.invokeSuspend(HttpTimeout.kt:149)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)\n02-16 22:44:13.811 E\/ACRA    (18522): \tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)\n02-16 22:44:13.811 D\/ACRA    (18522): Building report\n02-16 22:44:13.815 D\/ACRA    (18522): Starting collectors with priority FIRST\n02-16 22:44:13.816 D\/ACRA    (18522): Calling collector org.acra.collector.DropBoxCollector\n02-16 22:44:13.816 D\/ACRA    (18522): Calling collector org.acra.collector.LogCatCollector\n02-16 22:44:13.816 D\/ACRA    (18522): Collector org.acra.collector.DropBoxCollector completed\n02-16 22:44:13.817 D\/ACRA    (18522): Calling collector org.acra.collector.StacktraceCollector\n02-16 22:44:13.819 D\/ACRA    (18522): Collector org.acra.collector.StacktraceCollector completed\n02-16 22:44:13.826 D\/ACRA    (18522): Retrieving logcat output (buffer:default)...\n","CUSTOM_DATA":{},"INITIAL_CONFIGURATION":{"assetsSeq":0,"colorMode":5,"densityDpi":320,"fontScale":1,"hardKeyboardHidden":"HARDKEYBOARDHIDDEN_YES","keyboard":"KEYBOARD_NOKEYS","keyboardHidden":"KEYBOARDHIDDEN_NO","locale":"en_US","mcc":0,"mnc":0,"navigation":"NAVIGATION_DPAD","navigationHidden":"NAVIGATIONHIDDEN_NO","orientation":"ORIENTATION_LANDSCAPE","screenHeightDp":540,"screenLayout":"SCREENLAYOUT_SIZE_LARGE+SCREENLAYOUT_LONG_YES+SCREENLAYOUT_LAYOUTDIR_LTR+SCREENLAYOUT_ROUND_NO","screenWidthDp":960,"seq":4,"smallestScreenWidthDp":540,"touchscreen":"TOUCHSCREEN_NOTOUCH","uiMode":"UI_MODE_TYPE_TELEVISION+UI_MODE_NIGHT_NO","userSetLocale":false},"DEVICE_FEATURES":{"android.software.leanback_only":true,"android.software.adoptable_storage":true,"android.hardware.hdmi.cec":true,"android.software.backup":true,"com.amazon.amazonvideo.livingroom.feature.AV_TARGET":true,"android.software.activities_on_secondary_displays":true,"android.software.voice_recognizers":true,"android.software.picture_in_picture":true,"android.software.vulkan.deqp.level":true,"android.software.cant_save_state":true,"android.hardware.opengles.aep":true,"android.hardware.type.television":true,"android.hardware.bluetooth":true,"android.software.incremental_delivery":true,"droidlogic.software.netflix":true,"android.hardware.usb.host":true,"android.hardware.audio.output":true,"android.software.verified_boot":true,"android.hardware.gamepad":true,"android.software.live_tv":true,"nrdp.modelgroup":true,"android.software.autofill":true,"android.software.securely_removes_users":true,"android.software.leanback":true,"android.hardware.bluetooth_le":true,"android.software.app_widgets":true,"com.google.android.tv.installed":true,"android.software.input_methods":true,"android.hardware.vulkan.version":true,"android.hardware.wifi.passpoint":true,"android.hardware.screen.landscape":true,"com.google.android.feature.GLOBAL_ASSIST_TRIGGERING":true,"android.hardware.ram.normal":true,"android.software.webview":true,"android.hardware.camera.any":true,"android.hardware.vulkan.compute":true,"android.hardware.location.network":true,"android.software.cts":true,"android.software.app_enumeration":true,"android.hardware.wifi.direct":true,"android.software.ipsec_tunnels":true,"android.software.midi":true,"android.hardware.wifi":true,"android.hardware.location":true,"android.hardware.vulkan.level":true,"android.hardware.camera.external":true,"android.software.file_based_encryption":true,"glEsVersion":"3.2"},"USER_EMAIL":"N\/A","SHARED_PREFERENCES":{"default":{"store_version":7}},"IS_SILENT":false,"REPORT_ID":"961270bd-ac90-43b7-ae50-5e79d03b8667","BUILD":{"BOARD":"oppen","BOOTLOADER":"01.01.221026.231954","BRAND":"BenQ","CPU_ABI":"armeabi-v7a","CPU_ABI2":"armeabi","DEVICE":"YUL","DISPLAY":"20221114:user release-keys","FINGERPRINT":"BenQ\/QS02\/YUL:11\/RT\/20221114:user\/release-keys","HARDWARE":"amlogic","HOST":"ubuntu","ID":"RT","IS_DEBUGGABLE":false,"IS_EMULATOR":false,"MANUFACTURER":"SDMC","MODEL":"QS02","PERMISSIONS_REVIEW_REQUIRED":true,"PRODUCT":"QS02","RADIO":"unknown","SUPPORTED_32_BIT_ABIS":["armeabi-v7a","armeabi"],"SUPPORTED_64_BIT_ABIS":[],"SUPPORTED_ABIS":["armeabi-v7a","armeabi"],"TAGS":"release-keys","TIME":1668516221000,"TYPE":"user","UNKNOWN":"unknown","USER":"jiangruifeng","VERSION":{"ACTIVE_CODENAMES":[],"BASE_OS":"","CODENAME":"REL","INCREMENTAL":"20221114","PREVIEW_SDK_FINGERPRINT":"REL","PREVIEW_SDK_INT":0,"RELEASE":"11","RELEASE_OR_CODENAME":"11","SDK":"30","SDK_INT":30,"SECURITY_PATCH":"2022-10-05"}},"APP_VERSION_NAME":"0.0.0-dev.1","APP_VERSION_CODE":1,"USER_APP_START_DATE":"2024-02-16T22:35:57.078+01:00","BUILD_CONFIG":{"APPLICATION_ID":"org.jellyfin.androidtv.debug","BUILD_TYPE":"debug","DEBUG":true,"DEVELOPMENT":true,"VERSION_CODE":1,"VERSION_NAME":"0.0.0-dev.1"},"DISPLAY":{"0":{"currentSizeRange":{"smallest":[1080,1080],"largest":[1920,1920]},"flags":"FLAG_SUPPORTS_PROTECTED_BUFFERS+FLAG_SECURE","metrics":{"density":2,"densityDpi":320,"scaledDensity":"x2.0","widthPixels":1920,"heightPixels":1080,"xdpi":320,"ydpi":320},"realMetrics":{"density":2,"densityDpi":320,"scaledDensity":"x2.0","widthPixels":1920,"heightPixels":1080,"xdpi":320,"ydpi":320},"name":"Built-in Screen","realSize":[1920,1080],"rectSize":[0,0,1920,1080],"size":[1920,1080],"rotation":"ROTATION_0","isValid":true,"orientation":0,"refreshRate":59.940059661865234,"height":1080,"width":1920,"pixelFormat":1}},"INSTALLATION_ID":"18722c28-4112-44ee-a65b-d34b7fbc9a81","PACKAGE_NAME":"org.jellyfin.androidtv.debug","PHONE_MODEL":"QS02","ANDROID_VERSION":"11","BRAND":"BenQ","PRODUCT":"QS02","FILE_PATH":"\/data\/user\/0\/org.jellyfin.androidtv.debug\/files","USER_CRASH_DATE":"2024-02-16T22:44:13.875+01:00","CRASH_CONFIGURATION":{"assetsSeq":0,"colorMode":5,"densityDpi":320,"fontScale":1,"hardKeyboardHidden":"HARDKEYBOARDHIDDEN_YES","keyboard":"KEYBOARD_NOKEYS","keyboardHidden":"KEYBOARDHIDDEN_NO","locale":"en_US","mcc":0,"mnc":0,"navigation":"NAVIGATION_DPAD","navigationHidden":"NAVIGATIONHIDDEN_NO","orientation":"ORIENTATION_LANDSCAPE","screenHeightDp":540,"screenLayout":"SCREENLAYOUT_SIZE_LARGE+SCREENLAYOUT_LONG_YES+SCREENLAYOUT_LAYOUTDIR_LTR+SCREENLAYOUT_ROUND_NO","screenWidthDp":960,"seq":4,"smallestScreenWidthDp":540,"touchscreen":"TOUCHSCREEN_NOTOUCH","uiMode":"UI_MODE_TYPE_TELEVISION+UI_MODE_NIGHT_NO","userSetLocale":false},"DUMPSYS_MEMINFO":"","TOTAL_MEM_SIZE":12521615360,"AVAILABLE_MEM_SIZE":9631059968,"ENVIRONMENT":{"getDataDirectory":"\/data","getDataSystemDirectory":"\/data\/system","getDownloadCacheDirectory":"\/data\/cache","getExternalStorageDirectory":"\/storage\/emulated\/0","getExternalStorageState":"mounted","getInternalMediaDirectories":"[\/system\/media, \/oem\/media, \/product\/media]","getLegacyExternalStorageDirectory":"\/sdcard","getLegacyExternalStorageObbDirectory":"\/sdcard\/Android\/obb","getOdmDirectory":"\/odm","getOemDirectory":"\/oem","getProductDirectory":"\/product","getProductServicesDirectory":"\/product_services","getRootDirectory":"\/system","getStorageDirectory":"\/storage","getSystemExtDirectory":"\/system_ext","getVendorDirectory":"\/vendor","isExternalStorageEmulated":true,"isExternalStorageLegacy":false,"isExternalStorageManager":false,"isExternalStorageRemovable":false}}
```

--- a/app/src/main/java/org/jellyfin/androidtv/ui/browsing/BrowseGridFragment.java
+++ b/app/src/main/java/org/jellyfin/androidtv/ui/browsing/BrowseGridFragment.java
@@ -2,7 +2,6 @@ package org.jellyfin.androidtv.ui.browsing;
 
 import static org.koin.java.KoinJavaComponent.inject;
 
-import android.app.Activity;
 import android.os.Bundle;
 import android.os.Handler;
 import android.util.DisplayMetrics;
@@ -19,6 +18,7 @@ import android.widget.PopupWindow;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
 import androidx.leanback.widget.BaseGridView;
 import androidx.leanback.widget.OnItemViewClickedListener;
 import androidx.leanback.widget.OnItemViewSelectedListener;
@@ -85,7 +85,7 @@ public class BrowseGridFragment extends Fragment implements View.OnKeyListener {
     private final static int CHUNK_SIZE_MINIMUM = 25;
 
     private String mainTitle;
-    private Activity mActivity;
+    private FragmentActivity mActivity;
     private BaseRowItem mCurrentItem;
     private CompositeClickedListener mClickedListener = new CompositeClickedListener();
     private CompositeSelectedListener mSelectedListener = new CompositeSelectedListener();
@@ -763,7 +763,7 @@ public class BrowseGridFragment extends Fragment implements View.OnKeyListener {
         libraryPreferences.set(LibraryPreferences.Companion.getFilterUnwatchedOnly(), mAdapter.getFilters().isUnwatchedOnly());
         libraryPreferences.set(LibraryPreferences.Companion.getSortBy(), mAdapter.getSortBy());
         libraryPreferences.set(LibraryPreferences.Companion.getSortOrder(), getSortOption(mAdapter.getSortBy()).order);
-        CoroutineUtils.runBlocking((coroutineScope, continuation) -> libraryPreferences.commit(continuation));
+        CoroutineUtils.runOnLifecycle(getLifecycle(), (coroutineScope, continuation) -> libraryPreferences.commit(continuation));
     }
 
     private void addTools() {

--- a/app/src/main/java/org/jellyfin/androidtv/util/CoroutineUtils.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/util/CoroutineUtils.kt
@@ -8,12 +8,14 @@ import androidx.lifecycle.flowWithLifecycle
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
 import org.jellyfin.androidtv.constant.CustomMessage
 import org.jellyfin.androidtv.data.repository.CustomMessageRepository
 
-fun <T : Any> runBlocking(block: suspend CoroutineScope.() -> T) = kotlinx.coroutines.runBlocking {
-	block()
-}
+fun <T : Any> runOnLifecycle(
+	lifecycle: Lifecycle,
+	block: suspend CoroutineScope.() -> T
+) = lifecycle.coroutineScope.launch { block() }
 
 fun readCustomMessagesOnLifecycle(
 	lifecycle: Lifecycle,

--- a/app/src/main/java/org/jellyfin/androidtv/util/KeyProcessor.java
+++ b/app/src/main/java/org/jellyfin/androidtv/util/KeyProcessor.java
@@ -1,10 +1,12 @@
 package org.jellyfin.androidtv.util;
 
-import android.app.Activity;
 import android.view.Gravity;
 import android.view.KeyEvent;
 import android.view.MenuItem;
 import android.widget.PopupMenu;
+
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.LifecycleOwner;
 
 import org.jellyfin.androidtv.R;
 import org.jellyfin.androidtv.constant.CustomMessage;
@@ -54,11 +56,11 @@ public class KeyProcessor {
 
     private static String mCurrentItemId;
     private static BaseItemDto mCurrentItem;
-    private static Activity mCurrentActivity;
+    private static FragmentActivity mCurrentActivity;
     private static int mCurrentRowItemNdx;
     private static boolean isMusic;
 
-    public static boolean HandleKey(int key, BaseRowItem rowItem, Activity activity) {
+    public static boolean HandleKey(int key, BaseRowItem rowItem, FragmentActivity activity) {
         if (rowItem == null) return false;
         MediaManager mediaManager = KoinJavaComponent.<MediaManager>get(MediaManager.class);
         switch (key) {
@@ -179,7 +181,7 @@ public class KeyProcessor {
         return false;
     }
 
-    public static PopupMenu createItemMenu(BaseRowItem rowItem, UserItemDataDto userData, Activity activity) {
+    public static PopupMenu createItemMenu(BaseRowItem rowItem, UserItemDataDto userData, FragmentActivity activity) {
         BaseItemDto item = rowItem.getBaseItem();
         PopupMenu menu = new PopupMenu(activity, activity.getCurrentFocus(), Gravity.END);
         int order = 0;
@@ -269,7 +271,7 @@ public class KeyProcessor {
         return menu;
     }
 
-    private static void createPlayMenu(BaseItemDto item, boolean isMusic, Activity activity) {
+    private static void createPlayMenu(BaseItemDto item, boolean isMusic, FragmentActivity activity) {
         PopupMenu menu = new PopupMenu(activity, activity.getCurrentFocus(), Gravity.END);
         int order = 0;
         if (!isMusic && item.getType() != BaseItemKind.PLAYLIST) {
@@ -351,16 +353,16 @@ public class KeyProcessor {
                     });
                     return true;
                 case MENU_MARK_FAVORITE:
-                    toggleFavorite(true);
+                    toggleFavorite(mCurrentActivity, true);
                     return true;
                 case MENU_UNMARK_FAVORITE:
-                    toggleFavorite(false);
+                    toggleFavorite(mCurrentActivity, false);
                     return true;
                 case MENU_MARK_PLAYED:
-                    togglePlayed(true);
+                    togglePlayed(mCurrentActivity, true);
                     return true;
                 case MENU_UNMARK_PLAYED:
-                    togglePlayed(false);
+                    togglePlayed(mCurrentActivity, false);
                     return true;
                 case MENU_GOTO_NOW_PLAYING:
                     NavigationRepository navigationRepository = KoinJavaComponent.get(NavigationRepository.class);
@@ -387,10 +389,10 @@ public class KeyProcessor {
         }
     };
 
-    private static void togglePlayed(boolean played) {
+    private static void togglePlayed(LifecycleOwner lifecycleOwner, boolean played) {
         ItemMutationRepository itemMutationRepository = KoinJavaComponent.<ItemMutationRepository>get(ItemMutationRepository.class);
 
-        CoroutineUtils.runBlocking((scope, continuation) ->
+        CoroutineUtils.runOnLifecycle(lifecycleOwner.getLifecycle(), (scope, continuation) ->
                 itemMutationRepository.setPlayed(mCurrentItem.getId(), played, continuation)
         );
 
@@ -398,10 +400,10 @@ public class KeyProcessor {
         customMessageRepository.pushMessage(CustomMessage.RefreshCurrentItem.INSTANCE);
     }
 
-    private static void toggleFavorite(boolean favorite) {
+    private static void toggleFavorite(LifecycleOwner lifecycleOwner, boolean favorite) {
         ItemMutationRepository itemMutationRepository = KoinJavaComponent.<ItemMutationRepository>get(ItemMutationRepository.class);
 
-        CoroutineUtils.runBlocking((scope, continuation) ->
+        CoroutineUtils.runOnLifecycle(lifecycleOwner.getLifecycle(), (scope, continuation) ->
                 itemMutationRepository.setFavorite(mCurrentItem.getId(), favorite, continuation)
         );
 

